dir/path.go

```diff
@@ -79,21 +79,27 @@ const (
 
 var userConfigDir = os.UserConfigDir // for unit test
 
-func init() {
-	loadUserPath()
+// userConfigDirPath returns the user level {NOTATION_CONFIG} path.
+func userConfigDirPath() string {
+	if UserConfigDir == "" {
+		userDir, err := userConfigDir()
+		if err != nil {
+			// fallback to current directory
+			userDir = "."
```

Comment:
This will result in `./notation`. Would it be better to have `.notation` as `UserConfigDir` as a more Unix/Linux way?

Reply:
The current behaviour if `$HOME` is set would result in `$HOME/notation`. Would you rather change the notation folder be named `.notation` instead? So, if `$HOME` is set, the result would be `$HOME/.notation`. Or are you asking if `$HOME` is not set, the result should be `./.notation` otherwise the result should be `$HOME/notation`?

Reply:
It's not quite right. According to [os.UserConfigDir()](https://pkg.go.dev/os#UserConfigDir) doc,
> On Unix systems, it returns $XDG_CONFIG_HOME as specified by https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html if non-empty, else $HOME/.config. On Darwin, it returns $HOME/Library/Application Support. On Windows, it returns %AppData%. On Plan 9, it returns $home/lib.

If `$HOME` is set, the result would be `$HOME/.config/notation` on Linux.

I am asking if `$HOME` is not set, the result should be `./.notation`. Otherwise, the result should be `$HOME/.config/notation`.

Reply:
How about something like this?

```go
if UserConfigDir == "" {
	notationDir := notation
	userDir, err := userConfigDir()
	if err != nil {
		// fallback to current directory
		userDir = "."
		notationDir = fmt.Sprintf(".%s", notation) 
	}
	// set user config
	UserConfigDir = filepath.Join(userDir, notationDir)
}
```

Reply:
It's getting more complex. How about
```suggestion
			UserConfigDir = "." + notation
			return UserConfigDir
```